### PR TITLE
[MM-43339] Add Cloud Subscription Upgrade Confirmation Email

### PR DESCRIPTION
**Pull request by nickmisasi**
#### Summary
Adds an email to be sent to the customer when successfully upgrading their subscription. I've also left an accessor to this email in the webhooks, so we can trigger this from other systems in the future if necessary. 

image

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-43339

#### Release Note

```release-note
None
```

### CODE REVIEW

**Comment by johndavidlugtu** [comment]
@nickmisasi copy in body needs to be updated to: 

Your {workspace name} has now been upgraded. You will be billed starting {date}

We need to do away from referencing Cloud Starter, "charged based on number of enabled users".

**Comment by nickmisasi** [comment]
@johndavidlugtu updated
image

**Review by neallred — APPROVED** [review]
LGTM! A few nits and we probably need to run `make i18n-extract` to keep the linter happy.

File: app/cloud.go [hunk 1]
```diff
@@ -35,6 +37,46 @@ func (a *App) SendPaymentFailedEmail(failedPayment *model.FailedPayment) *model.
...
+
+	// Build readable trial end date
+	endTimeStamp := subscription.TrialEndAt
+	t := time.Unix(endTimeStamp, 0)
+	trialEndDate := fmt.Sprintf("%s %d, %d", t.Month(), t.Day(), t.Year())
+
+	// we want to at least have one email sent out to an admin
+	countNotOks := 0
+
+	for admin := range sysAdmins {
```
[the remark this answers is not in the document]
Comment (nickmisasi) [reply app/cloud.go:59]: Because copy + paste :P updated.

File: app/cloud.go [hunk 2]
```diff
@@ -35,6 +37,46 @@ func (a *App) SendPaymentFailedEmail(failedPayment *model.FailedPayment) *model.
...
+
+		err := a.Srv().EmailService.SendCloudUpgradeConfirmationEmail(sysAdmins[admin].Email, name, trialEndDate, sysAdmins[admin].Locale, *a.Config().ServiceSettings.SiteURL, subscription.GetWorkSpaceNameFromDNS())
+		if err != nil {
+			a.Log().Error("Error sending trial ended email to", mlog.String("email", sysAdmins[admin].Email), mlog.Err(err))
+			countNotOks++
+		}
+	}
+
+	// if not even one admin got an email, we consider that this operation errored
+	if countNotOks == len(sysAdmins) {
```
[the remark this answers is not in the document]
Comment (nickmisasi) [reply app/cloud.go:77]: Added a check on the length of the sysadmins and return early if its 0

**Review by johndavidlugtu — APPROVED** [review]

**Comment by neallred** [comment]
Also looks like `make email-mocks` needs to be run.

**Comment by nickmisasi** [comment]
@AGMETEOR Friendly reminder on this one :)

**Pull request merged**